### PR TITLE
Add project infrastructure

### PR DESCRIPTION
**Pull request by tjanez**
TODO:
- [x] Add golangci-lint configuration.
- [x] Add golangci-lint to ci-lint GitHub Actions workflow.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,10 @@
-# ledger-oasis-go
+# Oasis Core Ledger
```
Comment (kostko) [inline README.md:1]:
```suggestion
# Oasis Core Ledger Hardware Signer
```

Comment (kostko) [reply README.md:1]: Also maybe mention somewhere below what "ledger" refers to as it is a general term.

Comment (tjanez) [reply README.md:1]: I would prefer to keep the project's name in line with the repo's name. Also, the project's scope could be a bit wider than just providing a Ledger signer for Oasis Core.

I've tried to describe all this in the updated README. Please, take another look.

Comment (kostko) [reply README.md:1]: Looks good!

File: Makefile [hunk 2]
```diff
@@ -1,8 +1,59 @@
...
+	@$(ECHO) "$(CYAN)*** Checking go mod tidy...$(OFF)"
+	@$(ENSURE_GIT_CLEAN)
+	@$(CHECK_GO_MOD_TIDY)
+
+lint: $(lint-targets)
+
+# Test.
+test-targets := test-unit
+
+test-unit:
```
Comment (kostko) [inline Makefile:42]: What about the other test targets (e.g., we now have the `ledgerhw` tag for tests requiring actual hardware), the previous makefile also had other test targets but not sure if those actually did anything?

File: go.mod [hunk 3]
```diff
@@ -1,4 +1,4 @@
-module github.com/oasisprotocol/ledger-oasis-go
+module github.com/oasisprotocol/oasis-core-ledger
 
 go 1.13
```
Comment (kostko) [inline go.mod:3]: Should we bump this to 1.14 while we're here (oasis-core uses 1.14)?

Comment (tjanez) [reply go.mod]: Agreed. Done.

File: Makefile (hunk 2 above)
Comment (tjanez) [reply Makefile:42]: Yes, makes sense to try that. I'll add that in a separate PR.

**Pull request merged**